sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -351,11 +399,54 @@ export function getRawSqlParameters(parameters: SqlParameter[] | undefined): any
 /**
  * @internal
  * @ignore
- * Helper utility to build an instance of raw SQL parameter as `RawSqlParameter`
+ * Helper utility to extract array of userProperties key:value instances from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getRawUserProperties(parameters: { [key: string]: any } | undefined): any {
+  if (parameters == undefined) {
+    return undefined;
+  }
+
+  const rawParameters: RawKeyValuePair[] = [];
+  for (let [key, value] of Object.entries(parameters)) {
+    let type: string | number;
+    if (typeof value === "number") {
+      type = TypeMapForRequestSerialization.number;
+    } else if (typeof value === "string") {
+      type = TypeMapForRequestSerialization.string;
+    } else {
+      throw new TypeError(
+        `Unsupported type for the value in the user property {${key}:${JSON.stringify(value)}}`
+      );
+      // TODO: Add more cases if other types are supported.
+    }
+    const rawParameterValue: any = {};
+    rawParameterValue[Constants.XML_METADATA_MARKER] = {
+      "p4:type": type,
+      "xmlns:l28": "http://www.w3.org/2001/XMLSchema"
+    };
+    rawParameterValue[Constants.XML_VALUE_MARKER] = value;
+
+    const rawParameter: RawKeyValuePair = {
+      Key: key,
+      Value: rawParameterValue
+    };
+    rawParameters.push(rawParameter);
+  }
+  return {
+    KeyValueOfstringanyType: rawParameters
+  };
+}
+
+/**
+ * @internal
+ * @ignore
+ * Helper utility to build an instance of raw SQL parameter as `RawKeyValuePair`
  * from given `SqlParameter` input,
  * @param parameter parsed SQL parameter instance
  */
-function buildRawSqlParameter(parameter: SqlParameter): RawSqlParameter {
+function buildRawSqlParameter(parameter: SqlParameter): RawKeyValuePair {
```

Comment:
I think @ramya-rao-a had a question on this. 

You clearly want to call the type `RawSqlParameter` since you do it yourself (ie, the name of this method).  ;)

Reply:
I would instead make it `buildRawKeyValuePairFromSqlParameter`.

Reply:
Updated